apps/frontend/src/components/questionary/questionaryComponents/InstrumentPicker/QuestionaryComponentInstrumentPicker.tsx

```diff
@@ -216,7 +216,7 @@ export function QuestionaryComponentInstrumentPicker(
             value={
               Array.isArray(stateValue)
                 ? stateValue?.filter((i) => i).map((i) => i.instrumentId) || []
-                : stateValue?.instrumentId || '0'
+                : [stateValue?.instrumentId] || ['0']
```

Comment:
this expression is always truthy
```suggestion
                : stateValue?.instrumentId ? [stateValue.instrumentId] : []
```